### PR TITLE
V0 of confirm modal for build/snapshot/group approval

### PR DESCRIPTION
**Pull request by cadeParade**
## Description
The story of this feature is long, but here is a working, tested version that is not too unreasonable.

The goal was to make a central thing we could call with an API similar to `window.confirm` but based on Promises. There are a few complications to this, mostly due to promise rejection, but I've avoided most with this implementation

## Pictures
Look at this beaut
Build__12_-_org_without_plan_-_Percy___Visual_testing_as_a_service

## Reviewing this PR

_**Is Covered by Automated Tests?: Of course, who do you think I am**_

### CODE REVIEW

File: app/styles/_flash_messages.scss [hunk 1]
```diff
@@ -6,6 +6,30 @@ $flash-message-states: (
   "info": var(--light-gradient),
 );
 
+.confirm-container {
```
Comment (cadeParade) [inline app/styles/_flash_messages.scss]: Move this

File: tests/acceptance/build-test.js [hunk 2]
```diff
@@ -350,6 +352,88 @@ describe('Acceptance: Build', function() {
 
       await percySnapshot(this.test);
     });
+
+    it('blocks approval of snapshot if there are open review threads on snapshot', async function() { // eslint-disable-line
+      await BuildPage.visitBuild(urlParams);
+      await BuildPage.snapshots[0].clickApprove();
```
Comment (djones) [inline tests/acceptance/build-test.js]: there's quite heavy usage of `BuildPage.snapshots[0]`... you can decide but perhaps assign this as a `const`? It does make things slightly less cohesive though

File: tests/acceptance/build-test.js [hunk 3]
```diff
@@ -350,6 +352,88 @@ describe('Acceptance: Build', function() {
...
+          fingerprint: 'unapprovedGroup',
+        },
+      );
+
+      await BuildPage.visitBuild(urlParams);
+
+      await BuildPage.snapshotBlocks[0].clickApprove();
+      expect(BuildPage.confirmDialog.isVisible).to.equal(true);
+      expect(
+        BuildPage.snapshotBlocks[0].snapshotGroup.header.groupApprovalButton.isLoading,
```
Comment (djones) [inline tests/acceptance/build-test.js]: This is a long one. Consider setting up more direct to this button.

File: app/styles/_flash_messages.scss (hunk 1 above)
Comment (djones) [reply app/styles/_flash_messages.scss]: Perhaps: `app/styles/_modals.scss`

File: app/routes/organization/project/builds/build.js [hunk 4]
```diff
@@ -137,6 +152,27 @@ export default Route.extend({
...
+  },
+
+  _reviewConfirmMessage(snapshots, openReviewThreads) {
+    const numSnapshotsToApprove = snapshots.length;
+    const numOpenReviewThreads = openReviewThreads.length;
+    const snapshotString = pluralize(numSnapshotsToApprove, 'snapshot', {withoutCount: true});
+    const possessionString = numSnapshotsToApprove > 1 ? 'have' : 'has';
+    const commentString = pluralize(numOpenReviewThreads, 'comment', {withoutCount: true});
+
+    return `The ${snapshotString} you want to approve ${possessionString}
```
Comment (djones) [inline app/routes/organization/project/builds/build.js]:
```suggestion
    return `The ${snapshotString} you want to approve ${possessionString}
       unresolved ${commentString} requesting changes. Do you want to approve anyway?`;
```

**Pull request merged**